--- ./ssl/tls1.h.orig   2014-03-03 20:43:11.535295959 +0100
+++ ./ssl/tls1.h        2014-03-03 20:43:25.911430196 +0100
@@ -157,7 +157,7 @@
 extern "C" {
 #endif
 
-#define TLS1_ALLOW_EXPERIMENTAL_CIPHERSUITES   0
+#define TLS1_ALLOW_EXPERIMENTAL_CIPHERSUITES   1
 
 #define TLS1_2_VERSION                 0x0303
 #define TLS1_2_VERSION_MAJOR           0x03
--- ./ssl/s2_lib.c.orig 2014-03-05 18:16:40.966589866 +0100
+++ ./ssl/s2_lib.c      2014-03-05 18:17:36.572043680 +0100
@@ -122,7 +122,7 @@
 
 /* list of available SSLv2 ciphers (sorted by id) */
 OPENSSL_GLOBAL const SSL_CIPHER ssl2_ciphers[]={
-#if 0
+#if 1
 /* NULL_WITH_MD5 v3 */
        {
        1,
@@ -254,7 +254,7 @@
        168,
        },
 
-#if 0
+#if 1
 /* RC4_64_WITH_MD5 */
        {
        1,
